app/app.go

```diff
@@ -53,12 +57,14 @@ type App struct {
 
 var _ abci.Application = &App{}
 
-func NewApp(name string, ms sdk.CommitMultiStore) *App {
+// func NewApp(name string, ms sdk.CommitMultiStore) *App {
+func NewApp(name string, ms sdk.CommitMultiStore, keys map[string]sdk.SubstoreKey) *App {
```

Comment:
according to the above comment... storeKeys is something that NewMultiStore() should create for itself internally, not passed in.

Michael Sofaer also suggested that we create a StoreKey upon Mount.  But that's a separate Issue that we shouldn't address here.